corehq/apps/hqmedia/views.py

```diff
@@ -422,6 +422,15 @@ def valid_base_types(cls):
         return ['video']
 
 
+class ProcessTextFileUploadView(BaseProcessFileUploadView):
+    media_class = CommCareMultimedia
```

Comment:
One hacky thing you can do is make a new class but keep the `doc_type` as `CommCareMultimedia` - and then add some logic that knows how to convert things to the right class based on some other property. I took this approach when rolling out a new sync log format recently in case helpful. https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/phone/models.py#L679

(not blocking or necessary - just pointing out)